File: game.php
<?php

class Heroi {

    public $nome;
    public $idade;
    public $tipo;
    public $tipoAtaque="arma não definida";

    function __construct($nome,$idade,$tipo){
        $this->nome=$nome;
        $this->idade=$idade;
        $this->tipo=$tipo;
    }

    function heroiAtacar(){

        if($this->tipo=="mago"){
            $this->tipoAtaque="magia";
        }
        elseif($this->tipo=="guerreiro"){
            $this->tipoAtaque="espada";
        }
        elseif($this->tipo=="monge"){
            $this->tipoAtaque="artes marciais";
        }
        elseif($this->tipo=="ninja"){
            $this->tipoAtaque="shuriken";
        }

        echo "O heroi {$this->nome} do tipo {$this->tipo} atacou usando {$this->tipoAtaque}\n";
    }
}

$herois = [
    new Heroi("Goku",20,"monge"),
    new Heroi("Naruto",15,"ninja"),
    new Heroi("Samurai Shadow",33,"guerreiro")
];

foreach($herois as $h){
    $h->heroiAtacar();
}

File: game.cc
#include <iostream>
#include <vector>
using namespace std;

class Heroi{

public:
    string nome;
    int idade;
    string tipo;
    string tipoAtaque="arma não definida";

    Heroi(string n,int i,string t){
        nome=n;
        idade=i;
        tipo=t;
    }

    void heroiAtacar(){

        if(tipo=="mago")
            tipoAtaque="magia";
        else if(tipo=="guerreiro")
            tipoAtaque="espada";
        else if(tipo=="monge")
            tipoAtaque="artes marciais";
        else if(tipo=="ninja")
            tipoAtaque="shuriken";

        cout<<"O heroi "<<nome<<" do tipo "<<tipo<<" atacou usando "<<tipoAtaque<<endl;
    }
};

int main(){

    vector<Heroi> herois={
        Heroi("Goku",20,"monge"),
        Heroi("Naruto",15,"ninja"),
        Heroi("Samurai Shadow",33,"guerreiro")
    };

    for(auto &h:herois){
        h.heroiAtacar();
    }
}
